### PR TITLE
WIP - Dataframe docs

### PR DESCRIPTION
**Pull request by mrocklin**
This resets the dataframe docs to be more user focused

This also does various style cleaning

- normalize old pronouns `f` or `frame` to `df`
- Remove `names` in place of explicit names and `tokens`.  E.g. don't have names like `df-1` but rather descriptive names like `read-csv-1`

### CODE REVIEW

**Comment by mrocklin** [comment]
This touches a lot of code.  Merging fairly aggressively on this one.

File: dask/dataframe/shuffle.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def set_partition(df, index, divisions):
     dsk3 = {barrier_token: (barrier, list(dsk2))}
 
     # Collect groups
-    name = next(names)
+    name = 'set-partition--collet' + next(tokens)
```
Comment (cpcloud) [inline dask/dataframe/shuffle.py]: collect?

File: docs/source/dataframe.rst [hunk 2]
```diff
@@ -1,242 +1,135 @@
...
 
-``dask.dataframe.read_csv`` uses ``pandas.read_csv`` and so inherits all of
-that functions options.  Additionally it gains two new functionalities
+Internally a dask dataframe is split into many partitions, each partition is
+one pandas dataframe.  These dataframes are split vertically along the index.
+When our index is sorted and we know the values of the divisions of our
+partitions then we can be clever and efficient.
 
-1.  You can provide a globstring
+For example, if we have a time-series index then our partitoins might be divided by
```
Comment (cpcloud) [inline docs/source/dataframe.rst]: partitoins -> partitions

File: dask/dataframe/shuffle.py (hunk 1 above)
Comment (mrocklin) [inline dask/dataframe/shuffle.py]: Fixed

File: docs/source/dataframe.rst (hunk 2 above)
Comment (mrocklin) [inline docs/source/dataframe.rst]: Fixed

**Pull request merged**